crates/autopilot/src/solvable_orders.rs

```diff
@@ -162,21 +165,54 @@ impl SolvableOrdersCache {
         let mut invalid_order_uids = Vec::new();
         let mut filtered_order_events = Vec::new();
 
-        let orders = filter_banned_user_orders(db_solvable_orders.orders, &self.banned_users).await;
-        let removed = counter.checkpoint("banned_user", &orders);
-        invalid_order_uids.extend(removed);
+        let orders = {
+            let _timer = self
```

Comment:
I think having a separate function we could call like `self.stage_timer("banned_user_filtering")` would help with readability here.